source/StartConditions.cpp

```diff
@@ -231,28 +249,46 @@ const Conversation &StartConditions::GetConversation() const
 
 const Sprite *StartConditions::GetThumbnail() const noexcept
 {
-	return thumbnail;
+	// The state we ask for should always be present in the map, but just in case it isn't,
+	// make sure we're not accessing out of bound information.
+	auto it = infoByState.find(state);
+	return it == infoByState.end() ? nullptr : it->second.thumbnail;
+}
+
+
+
+const string &StartConditions::GetDisplayName() const noexcept
+{
+	auto it = infoByState.find(state);
+	static const string ILLEGAL = "(ILLEGAL STATE)";
+	return it == infoByState.end() ? ILLEGAL : it->second.name;
 }
 
 
 
-const std::string &StartConditions::GetDisplayName() const noexcept
+const string &StartConditions::GetDescription() const noexcept
 {
-	return name;
+	auto it = infoByState.find(state);
+	static const string ILLEGAL = "(ILLEGAL STATE)";
+	return it == infoByState.end() ? ILLEGAL : it->second.description;
 }
 
 
 
-const std::string &StartConditions::GetDescription() const noexcept
+const string &StartConditions::GetPlanetName() const noexcept
 {
-	return description;
+	auto it = infoByState.find(state);
+	static const string ILLEGAL = "(ILLEGAL STATE)";
+	return it == infoByState.end() ? ILLEGAL : it->second.planet;
 }
 
 
 
-const std::string &StartConditions::GetHint() const noexcept
+const string &StartConditions::GetSystemName() const noexcept
 {
-	return hint;
+	auto it = infoByState.find(state);
+	static const string ILLEGAL = "(ILLEGAL STATE)";
+	return it == infoByState.end() ? ILLEGAL : it->second.system;
```

Comment:
We're always guaranteed that the state exists in the map right? In that case, you can drop the ILLEGAL thing and just `infoByState[state]`.

Reply:
See the comment in the thumbnail function: checking whether we're out of bounds just in case. We might have a guarantee now, but are we sure someone won't mess this up in the future? :eyes:

Reply:
If we use `infoByState[state]` it'll silently create a new entry, and return an empty string. If someone does mess it up it'll get noticed (I hope lol).

Reply:
I'd prefer to return something rather than an empty string.